### PR TITLE
More hsStream cleanup

### PR DESCRIPTION
**Pull request by zrax**
* Always make bool I/O explicit, since we can stream both `bool`s and `BOOL`s (😢)
* Convert overloaded type readers and writers to a template
* Fix several cases of mismatched I/O type vs. its underlying storage.

### CODE REVIEW

File: Sources/Plasma/PubUtilLib/plDrawable/plDynaDecalMgr.cpp [hunk 1]
```diff
@@ -210,8 +210,8 @@ void plDynaDecalMgr::Read(hsStream* stream, hsResMgr* mgr)
         mgr->ReadKeyNotifyMe(stream, new plGenRefMsg(GetKey(), plRefMsg::kOnCreate, 0, kRefPartyObject), plRefFlags::kPassiveRef);
     }
 
-    fMaxNumVerts = (uint16_t)(stream->ReadLE32());
-    fMaxNumIdx = (uint16_t)(stream->ReadLE32());
+    fMaxNumVerts = stream->ReadLE32();
+    fMaxNumIdx = stream->ReadLE32();
```
Comment (Hoikas) [inline Sources/Plasma/PubUtilLib/plDrawable/plDynaDecalMgr.cpp:214]: FWIW, vertex and index buffers are limited to 16-bits. Not that I'm suggesting changing anything here, just making a note.

File: Sources/Plasma/PubUtilLib/plGImage/plDynamicTextMap.h [hunk 2]
```diff
@@ -187,8 +187,8 @@ class plDynamicTextMap : public plMipmap
 
         bool    MsgReceive(plMessage *msg) override;
 
-        uint16_t  GetVisibleWidth() { return fVisWidth; }
-        uint16_t  GetVisibleHeight() { return fVisHeight; }
+        uint32_t  GetVisibleWidth() { return fVisWidth; }
+        uint32_t  GetVisibleHeight() { return fVisHeight; }
```
Comment (Hoikas) [inline Sources/Plasma/PubUtilLib/plGImage/plDynamicTextMap.h:191]: While we're here...
```suggestion
        uint32_t  GetVisibleWidth() const { return fVisWidth; }
        uint32_t  GetVisibleHeight() const { return fVisHeight; }
```

File: Sources/Plasma/CoreLib/hsStream.h [hunk 3]
```diff
@@ -103,57 +103,42 @@ enum {
...
     void            WriteLEDouble(double value);
-    void            WriteLEDouble(int count, const double values[]);
+    void            WriteLEDouble(size_t count, const double values[]);
 
+    template <typename T>
+    inline void ReadLE(T* value)
+    {
+        Read(sizeof(T), value);
+        *value = hsToLE(*value);
+    }
```
Comment (Hoikas) [inline Sources/Plasma/CoreLib/hsStream.h:134]: Could we delete potentially bugprone instances of this, eg `ReadLE<size_t>` and `ReadLE<hsSsize_t>`?

Comment (zrax) [reply Sources/Plasma/CoreLib/hsStream.h:134]: Not really, since `size_t` may alias `uint32_t` on some environments (for example) :/

**Review by Hoikas — APPROVED** [review]

**Pull request merged**